_posts/2020-11-03-torch-tabular/torch_tabular.Rmd

```diff
@@ -0,0 +1,445 @@
+---
+title: "torch for tabular data"
+description: > 
+  How not to die from poisonous mushrooms. Also: How to use torch for deep learning on tabular data, including a mix of categorical and numerical features.
+
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanatorchtabular
+date: 11-03-2020
+categories:
+  - Torch
+  - R
+  - Tabular Data
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/preview.jpeg
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+Machine learning on image-like data can be many things: Fun (dogs vs. cats), societally useful (medical imaging), or societally harmful (surveillance). In comparison, tabular data -- the bread and butter of data science -- may seem more mundane.
+
+What's more, if you're interested in deep learning (DL) particularly, and looking for the extra benefits to be gotten from big data, big architectures, and big compute, you're much more likely to build an impressive showcase on the former instead of the latter.
+
+So for tabular data, why not just go with random forests, or gradient boosting, or other classical methods? I can think of at least a few reasons to learn about DL for tabular data:
+
+-   Even if all your features are interval-scale or ordinal, thus requiring "just" some form of (not necessarily linear) regression, applying DL may result in performance benefits due to sophisticated optimization algorithms, activation functions, layer depth, and more (plus interactions of all of these).
+
+-   If, in addition, there are categorical features, DL models may profit from *embedding* those in continuous space, discovering similarities and relationships that go unnoticed in one-hot encoded representations.
+
+-   What if most features are numeric or categorical, but there's also text in column F and an image in column G? With DL, different modalities can be worked on by different modules that feed their outputs into a common module, to take over from there.
+
+## Agenda
+
+In this introductory post, we keep the architecture straightforward. We don't experiment with fancy optimizers or nonlinearities. However, we do make use of embeddings, and pretty prominently at that. Thus from the above bullet list, we'll shed a light on the second, while leaving the other two for future posts.
```

Comment:
I had to scroll back up to see what the second item in that list was, so maybe give it a two or three word summary before the "…while leaving the other two for future posts"

Reply:
added a sentence on item (3), which i hadn't referred to before in that paragraph